lib/mongo_mapper/plugins/keys.rb

```diff
@@ -442,7 +442,7 @@ def load_from_database(attrs, with_cast = false)
       end
 
       def set_parent_document(key, value)
-        if key.type and value.instance_of?(key.type) && key.embeddable? && value.respond_to?(:_parent_document)
+        if key.type and value.kind_of?(key.type) && key.embeddable? && value.respond_to?(:_parent_document)
```

Comment:
`instance_of?` has been placed by https://github.com/mongomapper/mongomapper/commit/a60b04cc360f200b9f3c773150a53928192b2c8e#diff-f31cdda1ba47c5a9f26f34182b7dede3826f31e4652736e848dff0acc3019c46R302 for performance improvements.
However, its effect is a little as shown below.

```ruby
require "benchmark"

class A; end
class B < A; end
class C; end

Benchmark.bm(12) do |x|
  x.report("is_a?")        { 1000.times { B.is_a?(A) };        1000.times { C.is_a?(A) }; }
  x.report("instance_of?") { 1000.times { B.instance_of?(A) }; 1000.times { C.instance_of?(A) }; }
end
```

```
$ ruby ~/Desktop/benchmark.rb
                   user     system      total        real
is_a?          0.000094   0.000001   0.000095 (  0.000094)
instance_of?   0.000083   0.000001   0.000084 (  0.000083)
```